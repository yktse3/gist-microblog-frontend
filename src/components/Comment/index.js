import React from 'react';
import { string } from 'prop-types';
import {
  User,
  Content,
} from './styles';
import {
  Card,
} from '../Article/styles';

const Comment = ({ userID, content }) => (
  <Card>
    <User>{`${userID} replied:`}</User>
    <Content>{content}</Content>
  </Card>
);

Comment.propTypes = {
  userID: string.isRequired,
  content: string.isRequired,
};

export default Comment;
