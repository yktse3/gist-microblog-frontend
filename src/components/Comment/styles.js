import styled from 'styled-components';

export const User = styled.span`
  font-size: 16px;
  font-weight: 700;
`;

export const Content = styled.span`
  font-size: 13px;
  margin-top: 10px;
`;
