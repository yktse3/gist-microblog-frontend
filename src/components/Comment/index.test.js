import React from 'react';
import renderer from 'react-test-renderer';
import Comment from './index';

describe('Comment', () => {
  it('render successfully', () => {
    const component = renderer.create(
      <Comment
        userID="username"
        content="comment content"
        id="commentID"
      />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
