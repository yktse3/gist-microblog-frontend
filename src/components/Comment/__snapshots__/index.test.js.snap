// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment render successfully 1`] = `
<div
  className="sc-htpNat eCCHOJ"
>
  <span
    className="sc-bdVaJa hQekPs"
  >
    username replied:
  </span>
  <span
    className="sc-bwzfXH jmosDp"
  >
    comment content
  </span>
</div>
`;
